### PR TITLE
Added  Exception-Filters

### PR DESCRIPTION
**Pull request by wobo-apisero-shikhar**
### Exception Model:
-  Updated api-exception.model.ts and removed extra fields.

### Exception Filters:
- Added generic-exception.filter.ts file to handle any exceptions with status code other than 4XX.
- Added api-exception.filter.ts file to handle all exceptions with status code 4XX.

### Other:
- Added .gitignore file

### CODE REVIEW

File: src/shared/api-exception.model.ts [hunk 1]
```diff
@@ -1,24 +1,19 @@
...
   @ApiModelPropertyOptional()
   statusCode?: number;
 
   @ApiModelPropertyOptional()
   message?: string;
 
-  @ApiModelPropertyOptional()
-  status?: string;
-
   @ApiModelPropertyOptional()
```
Comment (wobo-apisero-vidhya) [inline src/shared/api-exception.model.ts:11]: Keep error as array

File: src/shared/filters/api-exception.filter.ts [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    constructor(private readonly httpAdapterHost: HttpAdapterHost) {}
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "You don't have access to the requested action. Please contact the admin.",
```
Comment (wobo-apisero-vidhya) [inline src/shared/filters/api-exception.filter.ts]: Please contact the admin. remove this in all messages.

File: src/shared/filters/api-exception.filter.ts [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    constructor(private readonly httpAdapterHost: HttpAdapterHost) {}
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "Conflict - Your request could not be processed.",
```
Comment (wobo-apisero-vidhya) [inline src/shared/filters/api-exception.filter.ts]: remove Conflict

File: src/shared/filters/api-exception.filter.ts [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+    constructor(private readonly httpAdapterHost: HttpAdapterHost) {}
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "Bad Request - Please check the request sent.",
```
Comment (wobo-apisero-vidhya) [inline src/shared/filters/api-exception.filter.ts]: Please remove - Bad Request. Http status code should define that. Message should be generic.

File: src/shared/filters/api-exception.filter.ts [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+    constructor(private readonly httpAdapterHost: HttpAdapterHost) {}
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "Authorization failed. Please contact the admin.",
```
Comment (wobo-apisero-vidhya) [inline src/shared/filters/api-exception.filter.ts]: pls remove -Please contact the admin.

File: src/shared/filters/api-exception.filter.ts [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "Your requested resource is no longer available and will not be available again.",
+          error: exception.message,
```
Comment (wobo-apisero-vidhya) [inline src/shared/filters/api-exception.filter.ts]: The requested resource is no longer available.

File: src/shared/api-exception.model.ts (hunk 1 above)
Comment (wobo-apisero-shikhar) [reply src/shared/api-exception.model.ts:11]: Done

File: src/shared/filters/api-exception.filter.ts (hunk 2 above)
Comment (wobo-apisero-shikhar) [reply src/shared/filters/api-exception.filter.ts]: done

File: src/shared/filters/api-exception.filter.ts (hunk 3 above)
Comment (wobo-apisero-shikhar) [reply src/shared/filters/api-exception.filter.ts]: done

File: src/shared/filters/api-exception.filter.ts (hunk 4 above)
Comment (wobo-apisero-shikhar) [reply src/shared/filters/api-exception.filter.ts]: done

File: src/shared/filters/api-exception.filter.ts (hunk 5 above)
Comment (wobo-apisero-shikhar) [reply src/shared/filters/api-exception.filter.ts]: done

File: src/shared/filters/api-exception.filter.ts (hunk 6 above)
Comment (wobo-apisero-shikhar) [reply src/shared/filters/api-exception.filter.ts]: done

**Review by wobo-apisero-vidhya — APPROVED** [review]

**Pull request merged**